OMEdit/OMEditGUI/Animation/Visualizer.cpp

```diff
@@ -504,55 +544,133 @@ void UpdateVisitor::apply(osg::Geode& node)
-    node.setStateSet(ss);
-    //set transparency
-    if (_shape.getTransparency())
-      makeTransparent(node, _shape.getTransparency());
-  }
+    changeColor(ss, _shape._color[0].exp, _shape._color[1].exp, _shape._color[2].exp);
+
+  //set transparency
+  makeTransparent(node, _shape.getTransparency());
 
+  node.setStateSet(ss);
   traverse(node);
 }
 
 /*!
- * \brief UpdateVisitor::makeTransparent
- * makes a geode transparent
- * \param event
+ * \brief UpdateVisitor::changeColor
+ * changes color for a geode
  */
-void UpdateVisitor::makeTransparent(osg::Geode& node, float transpCoeff)
+void UpdateVisitor::changeColor(osg::StateSet* ss, float r, float g, float b)
 {
-  node.getStateSet()->setMode( GL_BLEND, osg::StateAttribute::ON );
-  node.getStateSet()->setRenderingHint(osg::StateSet::TRANSPARENT_BIN);
   osg::Material *material;
-  if (NULL == node.getStateSet()->getAttribute(osg::StateAttribute::MATERIAL))
-  {
+  if (!ss->getAttribute(osg::StateAttribute::MATERIAL))
     material = new osg::Material();
+  else
+    material = dynamic_cast<osg::Material*>(ss->getAttribute(osg::StateAttribute::MATERIAL));
+  material->setDiffuse(osg::Material::FRONT, osg::Vec4f(r / 255, g / 255, b / 255, 1.0));
+  ss->setAttribute(material);
+}
+
+
+/*!
+ * \brief UpdateVisitor::applyTexture
+ * sets a texture for a geode
+ */
+void UpdateVisitor::applyTexture(osg::StateSet* ss, std::string imagePath)
+{
+  if (imagePath.compare(""))
+  {
+    osg::Image *image = nullptr;
+    std::string resIdent = ":/Resources";
+    if(!imagePath.compare(0,resIdent.length(),resIdent))
+    {
+      QImage* qim = new QImage(QString::fromStdString(imagePath));
+      image = convertImage(*qim);
+      image->setInternalTextureFormat(GL_RGBA);
+    }
+    else
+    {
+      image = osgDB::readImageFile(imagePath);
+    }
+    if (image)
+    {
+    osg::Texture2D *texture = new osg::Texture2D;
+    texture->setDataVariance(osg::Object::DYNAMIC);
+    texture->setFilter(osg::Texture::MIN_FILTER, osg::Texture::LINEAR_MIPMAP_LINEAR);
+    texture->setFilter(osg::Texture::MAG_FILTER, osg::Texture::LINEAR);
+    texture->setWrap(osg::Texture::WRAP_S, osg::Texture::CLAMP);
+    texture->setImage(image);
+    texture->setResizeNonPowerOfTwoHint(false);// dont output console message about scaling
+    ss->setTextureAttributeAndModes(0, texture, osg::StateAttribute::ON);
+    }
   }
   else
   {
-    material = dynamic_cast<osg::Material*>(node.getStateSet()->getAttribute(osg::StateAttribute::MATERIAL));
+    ss->getTextureAttributeList().clear();
+    ss->getTextureModeList().clear();
   }
-  material->setTransparency(osg::Material::FRONT_AND_BACK, transpCoeff);
-  node.getStateSet()->setAttributeAndModes(material, osg::StateAttribute::OVERRIDE);
 }
 
 
+osg::Image* UpdateVisitor::convertImage(const QImage& iImage)
+{
+   osg::Image* osgImage = new osg::Image();
+   if (false == iImage.isNull()) {
+      QImage glImage = QGLWidget::convertToGLFormat(iImage);
```

Comment:
You need to find a way to do it in Qt5 since QGLWidget is obsolete. From Qt documentation,
"This class is obsolete. It is provided to keep old source code working. We strongly advise against using it in new code."